File: resources/ts/schemas/HuggingFaceModelReference.ts
import { z } from "zod";

export const HuggingFaceModelReferenceSchema = z.object({
  filename: z.string(),
  repo_id: z.string(),
  revision: z.string(),
});

export type HuggingFaceModelReference = z.infer<
  typeof HuggingFaceModelReferenceSchema
>;

File: src/huggingface_model_reference.rs
use serde::Deserialize;
use serde::Serialize;

#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
#[serde(deny_unknown_fields)]
pub struct HuggingFaceModelReference {
    pub filename: String,
    pub repo_id: String,
    pub revision: String,
}
